applications/tari_console_wallet/src/ui/components/base_node.rs

```diff
@@ -42,9 +42,9 @@ impl BaseNode {
 impl<B: Backend> Component<B> for BaseNode {
     fn draw(&mut self, f: &mut Frame<B>, area: Rect, app_state: &AppState)
     where B: Backend {
-        let current_online_status = app_state.get_wallet_connectivity().get_connectivity_status();
+        let base_node_state = app_state.get_base_node_state();
```

Comment:
I'm hesitant to duplicate the online status in the base node state, the wallet connectivity service is the source of truth for the connectivity status and bugs may creep in making sure the value in the base node state is always in-sync with what the connectivity service is saying.

Reply:
I understand the hesitancy, though this is a draw method, we really should be using the variables from the state here.

Reply:
I agree, the `draw` method shouldn't be making calls into the backend it should be using the cached values in the AppState

Reply:
The call is using a watch (it's not async), so it simply returns the current set value - not sure if perf is the concern.

Reply:
Excuse me I misread this comment in the previous version of this PR. I thought this struct lived only in the console wallet. It definitely should not be duplicated in the Base Node Monitoring Service state, that really isn't needed.